File: PRACTICA1/JAVA/deepspace/src/deepspace/Weapon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Lucía Salamanca López
 *         Celia Arias Martínez
 */
class Weapon implements CombatElement{
   private String name;
   private WeaponType type;
   private int uses;
   
   Weapon(String nom, WeaponType tipo, int uso){
       name = nom;
       type = tipo;
       uses = uso;
   }
   
   Weapon(Weapon otro){
       name = otro.name;
       type = otro.type;
       uses = otro.uses;
   }
   
   public WeaponType getType(){
       return type;
   }
   @Override
   public int getUses(){
       return uses;
   }
   
   public float power(){
       return type.getPower();
   }
   @Override
   public float useIt(){
       if (uses > 0){
           uses--;
           return power();
       }
       return 1.0f;
   }
   
   public String toString(){
       String mensaje = "El arma se llama: "+name+", es de tipo: "+type+" y tiene "+uses+" usos.";
       return mensaje;
   }
 
   
   WeaponToUI getUIversion(){
       return new WeaponToUI(this); 
    }
}

File: PRACTICA1/RUBY/Deepspace/Weapon.rb
#encoding:utf-8
require './WeaponToUI'
module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class Weapon

  def initialize(name,type,uses)
    @name=name #string
    @type=type #WeaponType
    @uses=uses #int
  end

  def name
    @name
  end

  def type
    @type
  end

  def uses
    @uses
  end

  def self.newCopy(orig)
    new(orig.name, orig.type, orig.uses)
  end

  def getUIversion()
    WeaponToUI.new(self)
  end 

  def power
    @type.power()
  end



  def useIt

    if (@uses > 0)
      @uses=@uses-1
      return power
    else
      return 1.0
    end

  end

  def to_s
    "El arma se llama: "+@name+", es de tipo: "+@type.to_s+" y tiene "+@uses.to_s+" usos."
  end




end
end
